### ISSUE TITLE
Interactive Map Mouse Clicks

### ISSUE THREAD

**Issue by gsommers** [mapbox/mapbox-unity-sdk#168]
I'm working on building an interactive map in Unity as part of a database. I modified the slippy map example to allow users to zoom in/out and search for a location (like in the RasterTile example). I set the visualizer to use a custom map that I made on Mapbox Studio, and for the most part everything's displaying well.

My question is, how do I detect if the user clicks on a particular shapefile in my map style? My map contains shapefiles of tree distributions in Europe, and when a user clicks on a location, I want to display all of the tree species found at that location (along with information that I intend to pull from a text file).  I know how to detect a particular UnityTile using raycasts, but not the particular shapefiles displayed within those tiles.

I'm thinking using a custom map style might not work here, because it's really the _data_ that I want - i.e., the vector tile layers. However, from playing around with the mesh generation examples, although I've figured out how to detect what layers are currently visible and what properties they have for a certain tile, I don't know how to detect the properties at a specific point.

Basically what I want is something like the chloropleth map in this tutorial: https://www.mapbox.com/help/choropleth-studio-gl-pt-2/. I don't actually care about the color display, because the shapefile layers will be hidden underneath the basic map, but I want to mimic the functionality that detects when a user clicks on a state, except in Unity. (Since the map I'm making is part of a larger project, I'm pretty much confined to the Unity platform, as opposed to Mapbox GL JS.)

I've only been using Mapbox for a week, so any help for a beginner like me would be much appreciated! I'd be happy to upload files/scripts if that would clarify my question. Thanks!

**Comment by david-rhodes** [comment]
@gsommers Thanks for reaching out! 

It sounds like you have used the MapId in the `MapImageFactory`? If so, this simply generates a raster image (texture), which makes it difficult/impossible to extract metadata (such as geometric boundaries). 

For clarity, we don't directly support shapefiles with our Unity SDK. But, I'm assuming you've uploaded your data and generated a custom tileset? If so, use the Map Id from your tileset as the Id for a `MeshFactory`. Then, use the layer contained in that tileset for your `VectorLayerVisualizer` key. Once this is done you can add a `ColliderModifier` `GameObjectModifier` to your `Stack`. You can reference any of the mesh examples to get an idea of the general flow and factory structure.

When these meshes from your data are generated, a collider will also be created. You can use this to raycast from the camera to see what objects are being hit.

Hint: You can also use a `LayerModifier` to add these gameobjects to a specific layer to optimize your raycast with a `LayerMask`.

We will be blogging about a [demo we built for Unite Europe 2017](https://youtu.be/houwMdHj-v8?t=1s). This will demonstrate how to create meshes from vector tile data and add colliders to raycast against.

Does this answer your question?

**Comment by gsommers** [comment]
Yes, this answers my question. The `ColliderModifier` sounds like exactly what I was looking for. Thank you! I will close this issue.
